2696. Minimum String Length After Removing Substrings/solution.py

```diff
@@ -0,0 +1,29 @@
+class Solution:
+    # Given a string of uppercase letters, this function removes AB and CD strings as much
+    # as possible and then returns the length of the string. note that removing an AB or
+    # CD string might produce a new one.
+    def minLength(self, s: str) -> int:
+        # what is the index of the first character we are considering?
+
+        # iterate until the index is 2 characters from the length of the string
+
+            # if the pair of characters is AB or CD...
+
+                # remove the characters...
+
+                # and if this is not the first index...
+
+                    # decrement the index so that next iteration, we consider whether
+                    # it cleared the way for an AB or CD. 
+
+            # otherwise...
```

Comment:
I think you're over complicating this by trying to look at two characters each iteration so you have to handle the case if the 2nd character is an A or C. I think you should just iteration one character each time and do the same check. That would simplify things.